arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -559,21 +559,21 @@ static arangodb::ResultT<SyncerId> replicationSynchronize(
     std::shared_ptr<DatabaseTailingSyncer> tailingSyncer,
     VPackBuilder& sy) {
 
-  auto& vocbase = col->vocbase();
-  auto database = vocbase.name();
+  std::shared_ptr<TRI_vocbase_t> vocbase(&col->vocbase());
+  auto database = std::make_shared<std::string>(vocbase->name());
```

Comment:
We can make a copy of the name. No need to make a shared_ptr from it.
The copy can be passed by value or moved into the lambda.
```suggestion
  auto database = vocbase->name();
```